app/src/main/java/agency/five/codebase/android/movieapp/ui/home/HomeScreenViewModel.kt

```diff
@@ -0,0 +1,116 @@
+package agency.five.codebase.android.movieapp.ui.home
+
+import agency.five.codebase.android.movieapp.data.repository.MovieRepository
+import agency.five.codebase.android.movieapp.model.MovieCategory
+import agency.five.codebase.android.movieapp.ui.home.mapper.HomeScreenMapper
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class HomeScreenViewModel(
+    private val movieRepository: MovieRepository,
+    private val homeMapper: HomeScreenMapper
+) : ViewModel() {
+
+    private val popular = listOf(
+        MovieCategory.POPULAR_STREAMING,
+        MovieCategory.POPULAR_ON_TV,
+        MovieCategory.POPULAR_FOR_RENT,
+        MovieCategory.POPULAR_IN_THEATRES
+    )
+    private val nowPlaying = listOf(
+        MovieCategory.NOW_PLAYING_MOVIES,
+        MovieCategory.NOW_PLAYING_TV
+    )
+    private val upcoming = listOf(
+        MovieCategory.UPCOMING_TODAY,
+        MovieCategory.UPCOMING_THIS_WEEK
+    )
+
+    private val popularSelected = MutableStateFlow(MovieCategory.POPULAR_STREAMING)
+    private val nowPlayingSelected = MutableStateFlow(MovieCategory.NOW_PLAYING_MOVIES)
+    private val upcomingSelected = MutableStateFlow(MovieCategory.UPCOMING_TODAY)
+
+    private val initialHomeMovieCategoryViewState =
+        HomeMovieCategoryViewState(emptyList(), emptyList())
+
+    private val _popularCategoryViewState = MutableStateFlow(initialHomeMovieCategoryViewState)
+    val popularCategoryViewState = _popularCategoryViewState.asStateFlow()
+
+    private val _nowPlayingCategoryViewState = MutableStateFlow(initialHomeMovieCategoryViewState)
+    val nowPlayingCategoryViewState = _nowPlayingCategoryViewState.asStateFlow()
+
+    private val _upcomingCategoryViewState = MutableStateFlow(initialHomeMovieCategoryViewState)
+    val upcomingCategoryViewState = _upcomingCategoryViewState.asStateFlow()
+
+    init {
+        getPopularMovies()
+        getNowPlayingMovies()
+        getUpcomingMovies()
+    }
+
+    private fun getPopularMovies() {
+        viewModelScope.launch {
+            movieRepository.popularMovies(MovieCategory.POPULAR_STREAMING).collect { movies ->
```

Comment:
The problem is that you are not observing popularSelected (same problem with upcoming and nowPlaying), so it won't be updated when flow emits new value. Instead you can try to click on tab (This week for example) and then like or unlike a movie in that category, you will see that the selected tab changes now. It is because you are collecting (observing) popular movies which was changed, because one item was changed (from liked to unliked or vice versa) and now you collected new movies, but also the latest `popularSelected.value`. So you were updating it but your viewModle wasn't aware that this update happend until it read it again due to other update it was observing. 
So you would have to do something like this:
```
movieRepository.popMovies().collect { movies -> 
   popularSelected.collect { selected -> 
     ....
     }
     }
```
But you won't do this since we don't want to collect flow just to change value. We don't have to collect flow until we wish to do something with it (this will usually be in Composable function), we just have to change it and we have plenty of extension functions which can do this for us
So you will have to do soemthing like this:
``` selected.flatMapLatest { repo.movies().map{ ...}}.stateIn(...)``` Make sure to find out what is the difference between flatMapLatest and map, these are the most used `Flow` extensions.